tmt/steps/provision/podman.py

```diff
@@ -79,6 +79,27 @@ def wake(self) -> None:
         self.debug(
             f"Waking up container '{self.container}'.", level=2, shift=0)
 
+    def _setup_network(self) -> list[str]:
+        """
+        Set up the desired network.
+        Will look for existing network using the tmt workdir name,
+        or will create that network if it doesn't exist.
+        Returns the network arguments to be used in podman run command.
+        """
+
+        run_id = self._tmt_name().split('-')[1]
+        self.network = f"tmt-{run_id}-network"
+
+        try:
+            self.podman(Command('network', 'exists', self.network))
```

Comment:
With parallelized `provision` being a thing of the near future, /me hopes, checking for existence and taking actions based on this presents a race condition, as another thread may very well create the network between the `network exists` and `network create` commands in this thread.

Would it be possible to run `network create`, test the possible errors for "already exists", and re-raise every other exception? It's similar to the `network rm` below.

Reply:
handling it with error message checking now.